Code/Framework/AzFramework/AzFramework/Archive/Archive.cpp

```diff
@@ -1620,8 +1620,11 @@ namespace AZ::IO
         strBindRoot = !bindRoot.empty() ? bindRoot : szFullPath->ParentPath().Native();
 
         // Check if archive file disk exist on disk.
-        const bool pakOnDisk = FileIOBase::GetDirectInstance()->Exists(szFullPath->c_str());
-        if (!pakOnDisk && (nFactoryFlags & ZipDir::CacheFactory::FLAGS_READ_ONLY))
+
+        const bool archiveInAnotherArchive = FindPakFileEntry(szFullPath->Native());
+        const bool archiveLooseOnDisk = !archiveInAnotherArchive ? IO::FileIOBase::GetDirectInstance()->Exists(szFullPath->c_str()) : false;
+        const bool foundPak = archiveInAnotherArchive || archiveLooseOnDisk;
```

Comment:
Any reason to not call `IsFileExist` with a search parameter of `eFileLocation_Any`?